File: app/src/main/java/com/ballastlane/android/texasam/data/managers/remote/base/RestManagerPrivateService.java
package com.ballastlane.android.texasam.data.managers.remote.base;


import com.ballastlane.android.baselibrary.model.BaseServiceRest;
/**
 * @author Daniela Perez danielaperez@kogimobile.com on 5/26/17.
 */

public class RestManagerPrivateService extends BaseServiceRest<ApiServiceInterface.Private> {

    private static final String IMAGE_PART = "signature";
    private static final String MULTI_PART_FORM_DATA = "multipart/form-data";

    /**
     * @param apiServiceInterface
     * @deprecated
     */
    public RestManagerPrivateService(ApiServiceInterface.Private apiServiceInterface) {
        super(apiServiceInterface);
    }


//    public Observable<Void> sendDeviceToken(String devicetoken) {
//        return (getApiService()).sendDeviceToken(devicetoken)
//                .subscribeOn(Schedulers.io())
//                .observeOn(AndroidSchedulers.mainThread());
//    }
//
//    public Observable<Void> removeDeviceToken(String devicetoken) {
//        return (getApiService()).removeDeviceToken(devicetoken)
//                .subscribeOn(Schedulers.io())
//                .observeOn(AndroidSchedulers.mainThread());
//    }
//
//    public Observable<User> getUser() {
//        return (getApiService()).getUser()
//                .subscribeOn(Schedulers.io())
//                .observeOn(AndroidSchedulers.mainThread());
//    }



//    public Observable<RestUser> updateUser(UserRequest userRequest) {
//        return (getApiService()).updateUser(userRequest)
//                .subscribeOn(Schedulers.io())
//                .observeOn(AndroidSchedulers.mainThread());
//    }



}

File: sample/src/main/java/com/ballastlane/ballastlanebaseapp/data/managers/remote/RestManagerPrivateService.kt
package com.ballastlane.ballastlanebaseapp.data.managers.remote

import com.ballastlane.ballastlanebaseapp.data.repositories.user.source.RestUser
import io.reactivex.Single
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

/**
 * Created by Mariangela Salcedo (mariangelasalcedo@ballastlane.com) on 4/2/18.
 * Copyright (c) 2018 Ballast Lane Applications LLC. All rights reserved.
 */

class RestManagerPrivateService : BaseService<ApiServiceInterface>() {

    fun createUser(user: RestUser): Single<RestUser> {
        return getApiService()
                .createUser(user)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
    }

    fun updateUser(user: RestUser): Single<RestUser> {
        return getApiService()
                .updateUser(user)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
    }

}
